--- schema.sql	2013-10-02 13:12:24.000000000 +0000
+++ sqlite/schema.sql	2013-10-02 13:38:50.000000000 +0000
@@ -3,7 +3,7 @@
 
 CREATE TABLE dial_plans (
    phone_number TEXT NOT NULL PRIMARY KEY,
-   dial_plan    XML  NOT NULL
+   dial_plan    TEXT NOT NULL
 );
 
 -------------------------------------------------------------------------------
@@ -28,16 +28,16 @@
 INSERT INTO contact_types VALUES ('human'), ('function'), ('invisible');
 
 CREATE TABLE contacts (
-   id           INTEGER NOT NULL PRIMARY KEY, --  AUTOINCREMENT
+   id           INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    full_name    TEXT    NOT NULL,
    contact_type TEXT    NOT NULL REFERENCES contact_types (value)
 );
 
 CREATE TABLE organizations (
-   id        INTEGER NOT NULL PRIMARY KEY, --  AUTOINCREMENT
+   id        INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    full_name TEXT    NOT NULL,
    uri       TEXT    NOT NULL UNIQUE,
-   json      JSON    NOT NULL
+   json      TEXT    NOT NULL
 );
 
 CREATE INDEX organization_uri_index ON organizations (uri);
@@ -45,9 +45,9 @@
 CREATE TABLE organization_contacts (
    organization_id      INTEGER NOT NULL REFERENCES organizations (id) ON UPDATE CASCADE ON DELETE CASCADE,
    contact_id           INTEGER NOT NULL REFERENCES contacts (id) ON UPDATE CASCADE ON DELETE CASCADE,
-   wants_messages       BOOLEAN NOT NULL DEFAULT TRUE,
-   distribution_list_id INTEGER, --  Reference constraint added further down
-   attributes           JSON,
+   wants_messages       BOOLEAN NOT NULL DEFAULT 1,
+   distribution_list_id INTEGER NOT NULL REFERENCES distribution_lists (id) ON UPDATE CASCADE ON DELETE CASCADE,
+   attributes           TEXT,
 
    PRIMARY KEY (organization_id, contact_id)
 );
@@ -62,7 +62,7 @@
 INSERT INTO messaging_address_types VALUES ('e-mail'), ('sms');
 
 CREATE TABLE messaging_addresses (
-   id           INTEGER NOT NULL PRIMARY KEY, --  AUTOINCREMENT
+   id           INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    address      TEXT    NOT NULL,
    address_type TEXT    NOT NULL REFERENCES messaging_address_types (value),
 
@@ -73,8 +73,8 @@
    contact_id      INTEGER NOT NULL,
    organization_id INTEGER NOT NULL,
    address_id      INTEGER NOT NULL REFERENCES messaging_addresses (id),
-   confidential    BOOLEAN NOT NULL DEFAULT TRUE,
-   enabled         BOOLEAN NOT NULL DEFAULT FALSE,
+   confidential    BOOLEAN NOT NULL DEFAULT 1,
+   enabled         BOOLEAN NOT NULL DEFAULT 0,
    priority        INTEGER NOT NULL DEFAULT 0,
 
    PRIMARY KEY (contact_id, organization_id, address_id),
@@ -88,7 +88,7 @@
 INSERT INTO recipient_visibilities VALUES ('to'), ('cc'), ('bcc');
 
 CREATE TABLE distribution_lists (
-   id                      INTEGER NOT NULL PRIMARY KEY, --  AUTOINCREMENT
+   id                      INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    send_to_contact_id      INTEGER NOT NULL,
    send_to_organization_id INTEGER NOT NULL,
    recipient_visibility    TEXT    NOT NULL REFERENCES recipient_visibilities (value),
@@ -99,15 +99,6 @@
 );
 
 -------------------------------------------------------------------------------
---  Late reference from organization_contacts to distribution_lists:
-
-ALTER TABLE organization_contacts
-  ADD CONSTRAINT organization_contacts_distribution_list_id_foreign_key
-    FOREIGN KEY (distribution_list_id)
-      REFERENCES distribution_lists (id) MATCH SIMPLE
-      ON UPDATE CASCADE ON DELETE SET DEFAULT
-
--------------------------------------------------------------------------------
 --  System users:
 
 CREATE TABLE users (
@@ -124,3 +115,19 @@
 );
 
 -------------------------------------------------------------------------------
+--  Primary key indexes:
+
+CREATE INDEX dial_plans_primary_key ON dial_plans (phone_number);
+CREATE INDEX kinds_primary_key ON kinds (id);
+CREATE INDEX special_days_primary_key ON special_days (kind, day);
+CREATE INDEX contacts_primary_key ON contacts (id);
+CREATE INDEX organizations_primary_key ON organizations (id);
+CREATE INDEX organization_contacts_primary_key ON organization_contacts (organization_id, contact_id);
+CREATE INDEX address_types_primary_key ON messaging_address_types (value);
+CREATE INDEX end_points_primary_key ON messaging_end_points (contact_id, organization_id, address, address_type);
+CREATE INDEX recipient_visibilities_primary_key ON recipient_visibilities (value);
+CREATE INDEX distribution_lists_primary_key ON distribution_lists (id);
+CREATE INDEX users_primary_key ON users (name);
+CREATE INDEX user_ids_primary_key ON user_ids (openid);
+
+-------------------------------------------------------------------------------
