--- test_data.sql	2013-10-02 13:59:30.000000000 +0000
+++ sqlite/test_data.sql	2013-10-02 13:59:46.000000000 +0000
@@ -49,17 +49,17 @@
                                   address_id,
                                   confidential, enabled)
 VALUES --  Adaheads
-       (1, 1, 1, FALSE, TRUE),
-       (2, 1, 2, FALSE, FALSE),
-       (3, 4, 3, FALSE, TRUE),
-       (4, 4, 4, FALSE, TRUE),
-       (5, 4, 5, TRUE,  FALSE),
+       (1, 1, 1, 0, 1),
+       (2, 1, 2, 0, 0),
+       (3, 4, 3, 0, 1),
+       (4, 4, 4, 0, 1),
+       (5, 4, 5, 1,  0),
        --  Fishermans Friends                
-       (6, 1, 6, FALSE, TRUE),
-       (7, 1, 7, FALSE, FALSE),
+       (6, 1, 6, 0, 1),
+       (7, 1, 7, 0, 0),
        --  Responsum
-       (8, 1, 8, FALSE, TRUE),
-       (9, 2, 9, FALSE, TRUE);
+       (8, 1, 8, 0, 1),
+       (9, 2, 9, 0, 1);
 
 INSERT INTO distribution_lists (id,
                                           send_to_contact_id, send_to_organization_id,
@@ -84,10 +84,10 @@
 VALUES ('+4521490804', '<?xml version="1.0"?> <!DOCTYPE dial-plan> <dial-plan title="Jacob: Hang up on anonymous callers"> <start do="Start"/> <decision-tree title="Start"> <branch> <conditions> <caller number=""/> </conditions> <action do="Hang up"/> </branch> <fall-back do="Pass through"/> </decision-tree> <end-point title="Hang up"> <hang-up/> </end-point> <end-point title="Pass through"> <redirect to="+45 21 49 08 04"/> </end-point> </dial-plan>');
 
 INSERT INTO users (name, is_receptionist, is_service_agent, is_administrator)
-VALUES ('Tux',                      FALSE, FALSE, TRUE),
-       ('AdaHeads Test User One',   TRUE,  FALSE, FALSE),
-       ('AdaHeads Test User Two',   TRUE,  TRUE,  FALSE),
-       ('AdaHeads Test User Three', TRUE,  TRUE,  TRUE);
+VALUES ('Tux',                      0, 0, 1),
+       ('AdaHeads Test User One',   1, 0, 0),
+       ('AdaHeads Test User Two',   1, 1, 0),
+       ('AdaHeads Test User Three', 1, 1, 1);
 
 INSERT INTO user_ids (name, openid, priority)
 VALUES ('Tux','https://tux.myopenid.com/', 1), 
